architecture/2025-04-compute-platform-extensibility.md

```diff
@@ -0,0 +1,505 @@
+
+    switch env.Properties.Compute.Kind {
+    case rpv1.KubernetesComputeKind:
+        kubeProp := &env.Properties.Compute.KubernetesCompute
+
+        if kubeProp.Namespace == "" {
+            return renderers.EnvironmentOptions{}, errors.New("kubernetes' namespace is not specified")
+        }
+        envOpts.Namespace = kubeProp.Namespace
+
+    case rpv1.ACIComputeKind:
+        c := &env.Properties.Compute.ACICompute
+        if c.ResourceGroup == "" {
+            c.ResourceGroup = env.Properties.Providers.Azure.Scope
+        }
+        if c.ResourceGroup == "" {
+            return renderers.EnvironmentOptions{}, errors.New("resource group is not specified")
+        }
+        envOpts.Compute = &env.Properties.Compute
+
+    default:
+        return renderers.EnvironmentOptions{}, fmt.Errorf("%s is unsupported", env.Properties.Compute.Kind)
+    }
+```
+
+```go
+pkg/corerp/api/v20231001preview/application_conversion.go: (98)
+func fromAppExtensionClassificationDataModel(e datamodel.Extension) ExtensionClassification {
+    switch e.Kind {
+    case datamodel.KubernetesMetadata:
+        var ann, lbl = fromExtensionClassificationFields(e)
+        return &KubernetesMetadataExtension{
+            Kind:        to.Ptr(string(e.Kind)),
+            Annotations: *to.StringMapPtr(ann),
+            Labels:      *to.StringMapPtr(lbl),
+        }
+    case datamodel.KubernetesNamespaceExtension:
+        return &KubernetesNamespaceExtension{
+            Kind:      to.Ptr(string(e.Kind)),
+            Namespace: to.Ptr(e.KubernetesNamespace.Namespace),
+        }
+    case datamodel.ACIExtension:
+        return &AzureContainerInstanceExtension{
+            Kind:          to.Ptr(string(e.Kind)),
+            ResourceGroup: to.Ptr(e.AzureContainerInstance.ResourceGroup),
+        }
+    }
+
+    return nil
+}
+```
+
+```go
+pkg/corerp/frontend/controller/environments/createorupdateenvironment.go (98)
+
+    if newResource.Properties.Compute.Kind == rpv1.ACIComputeKind {
+        if err := e.createOrUpdateACIEnvironment(ctx, newResource); err != nil {
+            return nil, err
+        }
+    }
+```
+
+<!--
+High level description of updates to each component. Provide information on 
+the specific sub-components that will be updated, for example, controller, processor, renderer,
+recipe engine, driver, to name a few.
+
+#### UCP (if applicable)
+#### Bicep (if applicable)
+#### Deployment Engine (if applicable)
+#### Core RP (if applicable)
+#### Portable Resources / Recipes RP (if applicable)
+-->
+
+### Error Handling
+
+The approach to error handling will stay the same, with two additional features:
+
+## Test plan
+
+A basic test scenario will be added to Radius that will be included in the cloud functional tests.
+
+## Security
+
+The Radius [Secret Store](https://docs.radapp.io/reference/resource-schema/core-schema/secretstore/) portable type will create an Azure KeyVault when deploying to ACI.
+
+## Compatibility
```

Comment:
We should add an entry for how we keep the SDK up to date. We can't use the public SDK so we'd have to generate the SDK based off the swagger specs and would need some way to keep the specs up to date+compatible

Reply:
I added a bullet point to this section to cover the SDK integration.